test/bats/test.bats

```diff
@@ -42,10 +42,38 @@ teardown_file() {
   wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl wait --for condition=established --timeout=60s crd/constrainttemplates.templates.gatekeeper.sh"
 }
 
+@test "mutation crds are established" {
+  if [ -z $ENABLE_MUTATION_TESTS ]; then
+    skip "skipping mutation tests"
+  fi
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl wait --for condition=established --timeout=60s crd/assign.mutations.gatekeeper.sh"
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl wait --for condition=established --timeout=60s crd/assignmetadata.mutations.gatekeeper.sh"
+}
+
 @test "waiting for validating webhook" {
   wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration"
 }
 
+@test "gatekeeper mutation test" {
+  if [ -z $ENABLE_MUTATION_TESTS ]; then
+    skip "skipping mutation tests"
+  fi
+
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl apply -f ${BATS_TESTS_DIR}/mutations/k8sownerlabel_assignmetadata.yaml"
+  kubectl create namespace mutate-ns
```

Comment:
Can we use configmaps (or secret or another lightweight resource) for this, in a playground namespace?

This should help with test runtime/flakiness as creating namespaces is fairly heavy weight as it takes time to create the default service accounts and whatever other bootstrapping they require, and then to tear them down again.

It can also make test cleanup a bit easier, as we can simply delete the playground namespaces.

Reply:
I can certainly do that. I wanted to use a new namespace because I wanted to use separate resources/namespaces to test mutation to make sure theres no impact on the other validation tests. NOTE this mutate-ns is deleted right after the  assign test is done.

Reply:
+1 to the idea of using a new namespace, could it be a new playground namespace though? The speed/flakiness issues are my biggest concern.

Generally speaking, if our e2e tests are interacting with each other, that's not good, but we can look at that later.

Reply:
I have changed the test to mutate on a configmap instead of namespace. PTAL